src/openzaak/components/zaken/api/validators.py

```diff
@@ -280,6 +281,94 @@ def __call__(self, attrs: dict):
                 raise serializers.ValidationError(self.message, code=self.code)
 
 
+class DeelzaakReopenValidator:
+    """
+    Validate that the hoofdzaak is not closed  when a deelzaak is reopened.
+    """
+
+    code = "hoofdzaak-closed"
+    message = (
+        "De hoofdzaak is afgesloten."
+        "Een deelzaak kan pas worden geopend als de hoofzaak dat al is."
+    )
+
+    def __call__(self, attrs: dict):
+        if attrs.get("__is_eindstatus"):
+            return
+
+        zaak = attrs.get("zaak")
+        if not zaak or not zaak.hoofdzaak or not zaak.hoofdzaak.current_status:
+            return
+
+        if zaak.hoofdzaak.current_status.statustype.is_eindstatus():
+            raise serializers.ValidationError(self.message, code=self.code)
+
+
+class EndStatusDeelZakenValidator:
+    """
+    Validate that related deelzaken are closed when the end status is
+    being set.
+
+    The serializer sets the __is_eindstatus attribute in the data dict as
+    part of the ``to_internal_value`` method.
+    """
+
+    code = "deelzaken-closed"
+    message = (
+        "Er zijn gerelateerde deelzaken die nog niet zijn afgesloten zijn."
+        "Deze deelzaken moeten eerst afgesloten worden voordat de zaak afgesloten kan worden."
+    )
+
+    def __call__(self, attrs: dict):
+        if not attrs.get("__is_eindstatus"):
+            return
+
+        zaak = attrs.get("zaak")
+        # earlier validation failed possibly
+        if not zaak:
```

Comment:
I think the validation can also be skipped if the zaak has no deelzaken, right?
```suggestion
        if not zaak or not zaak.deelzaken.exists():
```